gkShell/gk.sh

```diff
@@ -0,0 +1,507 @@
+      vbSharedFolder remove "${HOST_GK_CONFIG}" "${SHARED_FOLDER_GKCONFIG}"
+  fi
+  if [ -d "${SCRIPTS_DIR}" ]; then
+      if hasScriptsMapping; then
+          vbSharedFolder remove "${HOST_GK_SCRIPTS}" "${SHARED_FOLDER_GKSCRIPTS}"
+      fi
+  fi
+  if ${hasStopped}; then
+    docker-machine start ${DOCKER_DEFAULT_MACHINE}
+  fi
+}
+command-help-build() {
+ echo " to build geokrety docker image ${IMAGE_FULLNAME}"
+ echo " this could takes some time but you have to execute only one time"
+ echo " show your images with \"$0 images\""
+}
+command-build() {
+  echo "docker build -t ${IMAGE_FULLNAME} ."
+  ${DOCKER_CMD} build -t ${IMAGE_FULLNAME} .
+}
+command-help-images() {
+ echo " show geokrety docker images"
+ echo " to build your image : \"$0 build\""
+}
+command-images() {
+  echo "docker images ${IMAGE_NAME}"
+  ${DOCKER_CMD} images ${IMAGE_NAME}
+}
+command-help-history() {
+ echo " show commands history for geokrety docker images"
+}
+image-history() {
+  echo "docker history ${IMAGE_FULLNAME}"
+  ${DOCKER_CMD} history ${IMAGE_FULLNAME}
+}
+# command-run-port-only() {
+#   echo "docker run -tid --name ${CONTAINER_NAME} -p 80:80 ${IMAGE_FULLNAME}"
+#   docker run -tid --name ${CONTAINER_NAME} -p 80:80 ${IMAGE_FULLNAME}
+# }
+# command-run-volume() {
+#   echo "docker run -tid --name ${CONTAINER_NAME} -p 80:80 --mount source=${VOLUME_NAME},target=${MOUNT_TARGET} ${IMAGE_FULLNAME}"
+#   docker run -tid --name ${CONTAINER_NAME} -p 80:80 --mount source=${VOLUME_NAME},target=${MOUNT_TARGET} ${IMAGE_FULLNAME}
+# }
+command-help-run() {
+ echo " run geokrety containers from: geokrety docker image, and mariadb image"
+}
+command-run() {
+  if isWindowsHost; then
+    command-run-windows
+  else
+    command-run-linux
+  fi
+}
+command-run-windows() {
+  computedParams
+  if ! hasWebsiteMapping || ! hasConfigMapping; then
+    echo " X please run $0 install";
+    exit 1
+  fi
+  DOCKER_IP=$(docker-machine ip ${DOCKER_DEFAULT_MACHINE})
+  echo "docker run ${DB_CONTAINER_NAME}"
+  ${DOCKER_CMD} run -tid --name ${DB_CONTAINER_NAME} -p 3306:3306 \
+      -v /${SHARED_FOLDER_GKCONFIG}/mariadb:/docker-entrypoint-initdb.d \
+      -e "MYSQL_ROOT_PASSWORD=password" \
+      -e "TIME_ZONE=UTC" \
+      mariadb:10.3
+
+  DB_IP=$(${DOCKER_CMD} inspect --format '{{ .NetworkSettings.Networks.bridge.IPAddress }}' ${DB_CONTAINER_NAME})
+  echo "alias db:${DB_IP}"
+
+  echo "docker run ${ADM_CONTAINER_NAME}"
+  ${DOCKER_CMD} run -tid --name ${ADM_CONTAINER_NAME} -p 8880:8080 \
+      --add-host="db:${DB_IP}" \
+      adminer:4.6.2
+
+  echo "docker run ${CONTAINER_NAME}"
+  VOLUME_OPTS="-v /${SHARED_FOLDER_GKWEBSITE}:${CONTAINER_MOUNT_TARGET}"
+  VOLUME_OPTS="${VOLUME_OPTS} -v /${SHARED_FOLDER_GKCONFIG}:${CONTAINER_MOUNT_CONFIG}"
+  if [ -d "${SCRIPTS_DIR}" ]; then
+    VOLUME_OPTS="${VOLUME_OPTS} -v /${SHARED_FOLDER_GKSCRIPTS}:${CONTAINER_MOUNT_SCRIPTS}"
+  fi
+
+  ${DOCKER_CMD} run -tid --name ${CONTAINER_NAME} -p 80:80 ${VOLUME_OPTS} \
```

Comment:
All of that sequential docker run cmd would be so simpler using `docker-compose`? :/

Reply:
agree 👍

Reply:
done